### ISSUE TITLE
External links with "+" don't seem to work correctly.

### ISSUE THREAD

**Issue by theherk** [epwalsh/obsidian.nvim#477]
### 🐛 Describe the bug

Generally, external links show this symbol, "" when concealed. For some reason, when there is a "+" in the url path, it seems to break the detection that this is an external link. For example, this code:

```markdown
[test without](https://www.test.com)
[test with](https://httpstat.us/not+real)
```

yields:

```markdown
test without 
test with
```

That prevents following the link with `  Error  10:35:35 notify.error Obsidian.nvim Failed to resolve file 'https://httpstat.us/not+real'`

### Config

```lua
  {
    "epwalsh/obsidian.nvim",
    version = "*",
    dependencies = {
      "nvim-lua/plenary.nvim",
      -- Required pngpaste on MacOS. `brew install pngpaste`
    },
    keys = {
      { "<leader>nb", "<cmd>ObsidianBacklinks<cr>", desc = "backlinks" },
      { "<leader>nf", "<cmd>ObsidianQuickSwitch<cr>", desc = "find" },
      { "<leader>ni", "<cmd>ObsidianPasteImg<cr>", desc = "img" },
      { "<leader>njm", "<cmd>ObsidianTomorrow<cr>", desc = "tomorrow" },
      { "<leader>njt", "<cmd>ObsidianToday<cr>", desc = "today" },
      { "<leader>njy", "<cmd>ObsidianYesterday<cr>", desc = "yesterday" },
      { "<leader>nll", "<cmd>ObsidianLink<cr>", desc = "link" },
      { "<leader>nln", "<cmd>ObsidianLinkNew<cr>", desc = "new" },
      { "<leader>nn", "<cmd>ObsidianNew<cr>", desc = "new" },
      { "<leader>nr", "<cmd>ObsidianRename<cr>", desc = "rename" },
      { "<leader>ns", "<cmd>ObsidianSearch<cr>", desc = "search" },
      { "<leader>nt", "<cmd>ObsidianTags<cr>", desc = "tags" },
      { "<leader>nwm", "<cmd>ObsidianWorkspace main<cr>", desc = "main" },
      { "<leader>nwo", "<cmd>ObsidianWorkspace onedrive<cr>", desc = "onedrive" },
      { "<localleader>mx", function() require("obsidian.util").toggle_checkbox() end, desc = "complete" },
    },
    opts = {
      daily_notes = {
        alias_format = "%Y-%m-%d",
      },
      follow_url_func = function(url) vim.fn.jobstart({ "open", url }) end,
      note_id_func = function(title)
        -- Given "Note Name" will make ID `<timestamp>-note-name`.
        -- Filename is the same with `.md` suffix.
        local suffix = ""
        if title ~= nil then
          suffix = title:gsub(" ", "-"):gsub("[^A-Za-z0-9-]", ""):lower()
        else
          for _ = 1, 4 do
            suffix = suffix .. string.char(math.random(65, 90))
          end
        end
        return tostring(os.time()) .. "-" .. suffix
      end,
      workspaces = {
        {
          name = "main",
          path = "~/org/foam",
        },
        {
          name = "onedrive",
          path = onedrive .. "/foam-onedrive",
        },
      },
    },
  },
```

### Environment

```
ᐅ nvim --version
  nvim --headless -c 'lua require("obsidian").info()' -c q
NVIM v0.9.5
Build type: Release
LuaJIT 2.1.1703358377

   system vimrc file: "$VIM/sysinit.vim"
  fall-back for $VIM: "/opt/homebrew/Cellar/neovim/0.9.5/share/nvim"

Run :checkhealth for more info
Plugins:
  [obsidian.nvim (v3.4.0)] Commit SHA: be82bc75072bae50598f7c4fee0b33f635e69d5e
  [plenary.nvim] Commit SHA: 4f71c0c4a196ceb656c824a70792f3df3ce6bb6d
  [nvim-cmp] Commit SHA: 04e0ca376d6abdbfc8b52180f8ea236cbfddf782
  [telescope.nvim] Commit SHA: b744cf59752aaa01561afb4223006de26f3836fd
Tools:
  [rg] ripgrep 14.1.0
Environment:
  [OS] Darwin⏎
```

**Comment by epwalsh** [comment]
Hey @theherk, fixed in 497d605. By the way, the conceal character is configurable in the `ui` settings.

**Comment by theherk** [comment]
Heck ya. Thanks for the quick turnaround. I think the default conceal character is fine; I was using it to illustrate there was a visual cue indicating it wasn't being interpreted as an external link. Good to know, though. Love this plugin btw.
